README.md

```diff
@@ -1,39 +1,597 @@
+
+    #! Usage from Daru::IO
+    Daru::IO::Exporters::RDS.new(df, args).call
+
+    #! Usage from Daru::DataFrame
+    df.to_rds(args)
+    ```
+
+### SQL Exporter
+
+[(Go to Table of Contents)](#table-of-contents)
+
+Exports a **Daru::DataFrame** into an SQL table.
+
+- **Docs**: [rubydoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Exporters/SQL)
+- **Gem Dependencies**: `dbd-sqlite3`, `dbi` and `sqlite3` gems
+- **Other Dependencies**: SQL
+- **Usage**:
+    ```ruby
+    #! Partially require just SQL Exporter
+    require 'daru/io/exporters/sql'
+
+    #! Usage from Daru::IO
+    Daru::IO::Exporters::SQL.new(df, args).call
+
+    #! Usage from Daru::DataFrame
+    df.to_sql(args)
+    ```
+
+# Tweaking IO modules
+
+**Daru-IO** currently supports various Import / Export methods, as it can be seen from the above list. But the
+list is NEVER complete - there may always be specific use-case format(s) that you need very badly, but
+might not fit the needs of majority of the community. In such a case, don't worry - you can always tweak
+(aka monkey-patch) daru-io in your application. The architecture of `daru-io` provides a neater way of
+monkey-patching into **Daru::DataFrame** to support your unique use-case.
+
+ - **Adding new IO modules**
+
+    Say, your unique use-case is of YAML IO Modules. Here's how you can proceed with tweaking -
+
+    ```ruby
+    #! YAML Importer
+
+    require 'daru/io'
+
+    class Daru::IO::Importers::YAML < Daru::IO::Importers::Base
+      Daru::DataFrame.register_io_module :from_yaml, self
+
+      def initialize(path)
+        @path = path
+      end
+
+      def call
+        #! Your code to read the YAML file
+        #! and return a Daru::DataFrame
+      end
+    end
+
+    df = Daru::DataFrame.from_yaml('filename.yaml') 
+    # or,
+    df = Daru::IO::Importers::YAML.new('filename.yaml').call
+    ```
+
+    ```ruby
+    #! YAML Exporter
+
+    require 'daru/io'
+
+    class Daru::IO::Exporters::YAML < Daru::IO::Exporters::Base
+      Daru::DataFrame.register_io_module :to_yaml, self
+
+      def initialize(dataframe, path)
+        super(dataframe) #! Have a look at documentation of Daru::IO::Exporters::Base#initialize
+        @path = path
+      end
 
-    $ bundle
+      def call
+        #! Your code to write the YAML file
+        #! with the data in the @dataframe
+      end
+    end
 
-Or install it yourself as:
+    df = Daru::DataFrame.new(x: [1,2], y: [3,4])
 
-    $ gem install daru-io
+    df.to_yaml('dataframe.yml')
+    # or,
+    Daru::IO::Exporters::YAML.new(df, 'dataframe.yml')
+    ```
 
-## Usage
+- **Modify existing IO modules**
 
-TODO: Write usage instructions here
+    Behaviour of existing IO modules can also be changed according to your needs, similar to the above example.
+    For example, if the CSV Importer has to be tweaked with a faster processing gem, simply replace the above
+    (psuedo)code with `class Daru::IO::Importers::CSV`.
```

Comment:
Is it a good suggestion? I believe it is better to suggest something like
```ruby
class MyNamespace::CustomCSV < Daru::IO::Importers::CSV
  Daru::DataFrame.register_io_module :custom_csv, self
```
WDYT?..